internal/service/searchdeployment/model_search_deployment.go

```diff
@@ -37,13 +37,16 @@ func NewTFSearchDeployment(ctx context.Context, clusterName string, deployResp *
 		result.Timeouts = *timeout
 	}
 
-	specsList, diagnostics := types.ListValueFrom(ctx, SpecObjectType, newTFSpecsModel(deployResp.GetSpecs()))
+	specs := deployResp.GetSpecs()
+	if !allowMultipleSpecs && len(specs) > 1 {
+		specs = specs[:1]
+	}
```

Comment:
Given the API response in this example, I think we should verify with owning team the current assumption of 1 item in specs. Don't feel we should be adding logic on our side in this case, I am even doubting why we have schema validation on `specs`:
`
Validators: []validator.List{
	listvalidator.SizeAtMost(1),
	listvalidator.SizeAtLeast(1),
},
`

Reply:
+1 to this, @EspenAlbert can you loop me in in the conversation with the team? Thank you!